### PR TITLE
🚧 8Y24PK task: Upgrade the Hermes AgentPlane bridge protocol across the three approved repositories [202608170928-8Y24PK]

### PR DESCRIPTION
**Pull request by densmirnov**
Task: `202608170928-8Y24PK`
Title: Upgrade the Hermes AgentPlane bridge protocol across the three approved repositories
Canonical task record: `.agentplane/tasks/202608170928-8Y24PK/README.md`

## Summary

Upgrade the Hermes AgentPlane bridge protocol across the three approved repositories

Implement the user-approved plan for AgentPlane 0.7.6, agentplane-hermes-plugin 0.2.0, and current Hermes worker-lane dispatch. Scope roots are /Users/densmirnov/Github/agentplane, /Users/densmirnov/Github/agentplane-hermes-plugin, and /Users/densmirnov/Github/hermes-agent. Required effects include source, tests, docs, public API, schema, CI/release metadata, security boundary, network reads, hosted external writes, and publication through explicit authority. Prove PLANNER/approval/EXECUTOR/EVALUATOR, retry, stale-run, and terminal attestation without direct kanban.db writes. Existing D5MAJ3 and failed structured-intake DDW1J5 are superseded and must not be implemented or published.

## Scope

- In scope: Implement the user-approved plan for AgentPlane 0.7.6, agentplane-hermes-plugin 0.2.0, and current Hermes worker-lane dispatch. Scope roots are /Users/densmirnov/Github/agentplane, /Users/densmirnov/Github/agentplane-hermes-plugin, and /Users/densmirnov/Github/hermes-agent. Required effects include source, tests, docs, public API, schema, CI/release metadata, security boundary, network reads, hosted external writes, and publication through explicit authority. Prove PLANNER/approval/EXECUTOR/EVALUATOR, retry, stale-run, and terminal attestation without direct kanban.db writes. Existing D5MAJ3 and failed structured-intake DDW1J5 are superseded and must not be implemented or published.
- Out of scope: unrelated refactors not required for "Upgrade the Hermes AgentPlane bridge protocol across the three approved repositories".

## Verification

- State: ok
- Note: Verified: CLI-owned checks passed before independent EVALUATOR review.
- Canonical workflow state lives in the task README.

<details>
<summary>Raw evidence</summary>

- Updated: 2026-08-17T09:29:36.876Z
- Branch: task/202608170928-8Y24PK/upgrade-the-hermes-agentplane-bridge-protocol-ac
- Head: computed live by `agentplane pr check` / `agentplane integrate`

```text
 docs/recipes/hermes-agentplane.mdx                 |  18 +++-
 docs/workflow-guides/hermes-kanban.mdx             |  74 +++++++-------
 integrations/hermes-agentplane-plugin/README.md    |  77 ++++++++++-----
 .../lane-registry.example.json                     |  19 ++++
 .../protocol-v2.schema.json                        |  24 +++++
 .../src/commands/hermes/hermes-environment.ts      |  41 +++++++-
 .../src/commands/hermes/hermes-runtime.ts          |  35 ++++---
 .../src/commands/hermes/hermes.command.test.ts     | 108 ++++++++++++++++++++-
 .../src/commands/hermes/hermes.command.ts          |  48 ++++++++-
 9 files changed, 351 insertions(+), 93 deletions(-)
```

</details>

### CODE REVIEW

File: packages/agentplane/src/commands/hermes/hermes.command.ts [hunk 1]
```diff
@@ -354,8 +367,29 @@ export function makeRunHermesDoctorHandler(deps: {
...
+      allowed_roots_fail_closed: plugin.allowed_roots_fail_closed,
+    };
+    const installationReady = Object.values(installationChecks).every(Boolean);
+    const workerContextReady =
+      installationReady &&
+      Boolean(env.task_id && env.board && env.run_id && env.workspace && env.claim_lock_present);
     const payload = {
-      ok: registry.error === null,
+      ok: installationReady,
+      installation_ready: installationReady,
```
[the remark this answers is not in the document]
Comment (densmirnov) [reply packages/agentplane/src/commands/hermes/hermes.command.ts:393]: Fixed in 6fc98c17: `agentplane hermes doctor` now returns exit code 1 whenever the integration is not ready; the readiness test expectations were updated.

File: packages/agentplane/src/commands/hermes/hermes-environment.ts [hunk 2]
```diff
@@ -13,6 +15,43 @@ export function hermesEnvSnapshot() {
...
+    native_worker_lane_api: process.env.AGENTPLANE_HERMES_NATIVE_WORKER_LANE_API?.trim() === "1",
+    allowed_roots: allowedRoots,
+    allowed_roots_fail_closed: allowedRoots.length > 0,
+  };
+}
+
+export async function inspectCommandAvailability(command: string) {
+  const candidates = command.includes(path.sep)
+    ? [path.resolve(command)]
+    : normalizedPathList(process.env.PATH).map((entry) => path.join(entry, command));
```
[the remark this answers is not in the document]
Comment (densmirnov) [reply packages/agentplane/src/commands/hermes/hermes-environment.ts]: Fixed in 6fc98c17: command discovery now honors Windows PATHEXT and path delimiters, with injected platform/path dependencies and a focused regression test.

File: packages/agentplane/src/commands/hermes/hermes.command.ts [hunk 3]
```diff
@@ -309,6 +311,17 @@ async function runHermesLifecycleCommand(
       message: `--body is required for Hermes lifecycle action: ${action}`,
     });
   }
+  if (
+    !parsed.dryRun &&
+    (!env.task_id || !env.run_id || !env.claim_lock_present || !env.workspace)
+  ) {
```
[the remark this answers is not in the document]
Comment (densmirnov) [reply packages/agentplane/src/commands/hermes/hermes.command.ts]: Fixed in 6fc98c17: non-dry-run lifecycle mutations now fail closed unless HERMES_KANBAN_BOARD is configured, with a focused regression test.
